assets/queries/k8s/statefulsets_headless_service_name/query.rego

```diff
@@ -0,0 +1,32 @@
+package Cx
+
+CxPolicy [ result ] {
```

Comment:
You're not checking if the resource is of the kind 'StatefulSet', it might not be appropriate to generalize, especially since the query addresses StatefulSets specifically.

Reply:
Also, it seems that you check if a StatefulSet has a headless service in a separate Service resource. So you should do the following:
- check if the resource is of kind StatefulSet 
- check the labels matching
- validate the service that is associated
- check if that service is headless